articles/getting-started-with-jwt-using-angular8-and-nodejs/index.md

```diff
@@ -0,0 +1,271 @@
+    req = req.clone({ headers: req.headers.set('Accept', 'application/json') });
+ 
+    return next.handle(req)
+        .pipe(
+           catchError((error: HttpErrorResponse) => {
+                //Catching Error Stage
+                if (error && error.status === 401) {
+                    console.log("ERROR 401 UNAUTHORIZED")
+                }
+                const err = error.error.message || error.statusText;
+                return throwError(error);                    
+           })
+        );
+  }  
+}
+
+```
+### Interception Stage Code Explanation
+
+In authentication, the token obtained from the server is stored locally. Then will be retrieved from local storage and the header `httpRequest req` cloned and Authorisation, Bearer: token header is added into it. Then the token sent in the httpRequest header.
+
+### Error Stage Code Explanation
+
+In case of an error response or 401, 402, etc., error status, the pipe will help catch the error. The user will not authenticate due to a bad request or Unauthorized Request. The error in the call is returned to the frontend in case of further error requests.
+
+## Creating the Backend
+
+I will start by creating a directory in the base folder for the server and initialize it as a node project by running the following commands.
+```bash
+$ mkdir node_server
+$ cd node_server
+$ npm init –y
+```
+The necessary libraries ([express](https://www.npmjs.com/package/express), [body-parser](https://www.npmjs.com/package/body-parser), [express-jwt](https://www.npmjs.com/package/express-jwt)) are required to be installed. Install them by running the command below:
+```bash
+$ npm i -S express cors body-parser express-jwt jsonwebtoken
+```
+After running the command above, a new file `app.js` containing the code below is created.
+```javascript
+const express       = require('express')
+const bodyParser    = require('body-parser');
+const cors          = require('cors');
+
+const app           = express();
+const port          = 3000;
+
+app.use(cors());
+app.options('*', cors());
+app.use(bodyParser.json({limit: '10mb', extended: true}));
+app.use(bodyParser.urlencoded({limit: '10mb', extended: true}));
+
+app.get('/', (req, res) => {
+    res.json("Hello World");
+});
+
+/* more code to be added later */
+
+
+/* the listen function */
+app.listen(port, function() {
+    console.log("Listening to " + port);
+});
+```
+Next, I will create a route where a token is generated, and it will be where the user authenticates. On successful authentication, the token is sent. Below code is added to the file `app.js`.
+```javascript
+//SECRET FOR JWT
+let secret = 'some_secret';
+
+/* Create token to be used */
+app.get('/token/sign', (req, res) => {
+    var userData = {
+        "name": "My Name",
+        "id": "1234"
+    }
+    let token = jwt.sign(userData, secret, { expiresIn: '15s'})
+    res.status(200).json({"token": token});
+});
```

Comment:
I get a `ReferenceError: jwt is not defined` error when I test `/token/sign` using Postman. Looks like you forgot to import jwt.
```js
const jwt = require('jsonwebtoken')
```